trieste/acquisition/function.py

```diff
@@ -277,14 +312,28 @@ def prepare_acquisition_function(
         if len(dataset.query_points) == 0:
             raise ValueError("Dataset must be populated.")
 
-        gumbel_sampler = GumbelSampler(self._num_samples, model)
+        if not self._use_thompson:  # use Gumbel sampler
+            sampler: ThompsonSampler = GumbelSampler(
+                self._num_samples, model, sample_min_value=True
+            )
+        else:
```

Comment:
Clearer as a single if/elif/else block:

```suggestion
        elif self._num_fourier_features is not None:
```